packages/common-ts/src/base-service/base-service-v2.ts

```diff
@@ -406,29 +411,28 @@ export abstract class BaseServiceV2<
 
         // Sleep between loops if we're still running (service not stopped).
         if (this.running) {
```

Comment:
Hmm what is the intended behavior here? Unclear where `this.running` is set

Reply:
It gets set immediately on 398 before the service starts running and when server is told to shut down it gets set to false.   The reason we need this.running in addition to ending the polling promise is for one specific corner case of the server is told to stop while this.main on like 402 is in the middle of running.   It only exists to make the the polling still stops if that race condition happens.

Reply:
Now that I think about it, we could get rid of this.running and use this.loop as source of truth here. f Set this.loop to false when looping is supposed to end.[

Reply:
Currently this.loop is set in the user options and never changes.

Reply:
I got rid of this.running in favor of using this.loop in this commit https://github.com/ethereum-optimism/optimism/pull/2731/commits/f59c504d2e850963ee5cf9edf266e5085b203c4f

Reply:
I prefer having a separate `running` boolean. `this.loop` is supposed to represent whether or not the service is supposed to loop, `this.running` represents whether the service is currently running.